CMakeLists.txt

```diff
@@ -3,7 +3,7 @@ project(cpp-template VERSION 0.1.0)
 
 set(CMAKE_CXX_STANDARD 17)
 
-include_directories(include)
+include_directories(config controller hal)
```

Comment:
One solution to this problem is that we add the header itself in the CMakeLists.txt that it's used for creating the library.
For example:
in `src/config`:
```cmake
set(ConfigLibSources 
    Configuration.cpp
    Configuration.h
    )

add_library(ConfigLib ${ConfigLibSources})

```

This should fix it 😄

Reply:
You can remove this, the include directory is already been covered by the new changes you made

Reply:
> You can remove this, the include directory is already been covered by the new changes you made

After this change, it looks really great!